packages/liferay-npm-scripts/src/scripts/build.js

```diff
@@ -62,11 +62,44 @@ function runBundler() {
 	}
 }
 
-/**
- * Runs the `liferay-npm-bridge-generator` executable.
- */
+///
+// Runs the `liferay-npm-bridge-generator` executable and removes sourcemaps to
+// ensure `liferay-npm-scripts build` is idempotent.
+//
+// In our common workflow right now, runBundler and runBridge can interact to
+// create some kind of circular dependency where files keep being re-processed
+// over and over or are only processed after the first pass.
+//
+// 1st pass
+// - runBundler
+// - runBridge: will expand output with an additional `bridge` folder
+//
+// 2nd pass
+// - runBundler: will now see `bridge` folder to process generating sourcemaps
+// - runBridge: depending on the configuration it can re-process previous `bridge`
+// 				folders, so it's advised to always add `!**/bridge` inside the
+//				`.npmbridgerc` `file-globs` configuration.
+//
+// To ensure output is the same after 1st and 2nd pass, we need to cleanup rogue
+// sourcemaps in the bridge-generator output folder.
+//
+// See https://github.com/petershin/liferay-portal/pull/724 for more details.
+//
 function runBridge() {
 	spawnSync('liferay-npm-bridge-generator');
+
+	// Retrieves the `.npmbridgerc` configuration which we already know exists
+	const bridgeConfig = JSON.parse(
+		fs.readFileSync(path.join(CWD, '.npmbridgerc'), 'utf8')
+	);
+
+	Object.keys(bridgeConfig).forEach(moduleKey => {
+		const output = bridgeConfig[moduleKey].output;
+
+		if (output) {
+			spawnSync('rimraf', [`${output}/**/*.map`]);
```

Comment:
I don't know why we're using `spawnSync` here instead of calling the `rimraf.sync` API directly (I know the one other place in this project that we use `rimraf`, we do it in the same way, but I am not sure there is a good reason for it).